openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -306,28 +315,27 @@ private static String methodName(String operation, PathItem.HttpMethod method) {
     private OpenAPI prepareModel(OpenApiConfig config, OpenApiStaticFile staticFile,
             List<? extends IndexView> filteredIndexViews) {
         try {
-            synchronized (OpenApiDocument.INSTANCE) {
-                OpenApiDocument.INSTANCE.reset();
-                OpenApiDocument.INSTANCE.config(config);
-                OpenApiDocument.INSTANCE.modelFromReader(OpenApiProcessor.modelFromReader(config, getContextClassLoader()));
-                if (staticFile != null) {
-                    OpenApiDocument.INSTANCE.modelFromStaticFile(OpenAPIParser.parse(helper().types(), staticFile.getContent(),
-                            OpenAPIMediaType.byFormat(staticFile.getFormat())));
-                }
-                if (isAnnotationProcessingEnabled(config)) {
-                    expandModelUsingAnnotations(config, filteredIndexViews);
-                } else {
-                    LOGGER.log(Level.FINE, "OpenAPI Annotation processing is disabled");
-                }
-                OpenApiDocument.INSTANCE.filter(OpenApiProcessor.getFilter(config, getContextClassLoader()));
-                OpenApiDocument.INSTANCE.initialize();
-                OpenAPIImpl instance = OpenAPIImpl.class.cast(OpenApiDocument.INSTANCE.get());
-
-                // Create a copy, primarily to avoid problems during unit testing.
-                // The SmallRye MergeUtil omits the openapi value, so we need to set it explicitly.
-                return MergeUtil.merge(new OpenAPIImpl(), instance)
-                        .openapi(instance.getOpenapi());
+            // The write lock guarding the model has already been acquired.
```

Comment:
I wish we could use `assert` so you could assert this.

Reply:
Noted.